src/utils/parser/getSymbols.js

```diff
@@ -84,9 +90,217 @@ export default function getSymbols(source: SourceText): SymbolDeclarations {
+      }
+
+      if (t.isMemberExpression(path)) {
+        const { start, end } = path.node.property.loc;
+        symbols.memberExpressions.push({
+          name: path.node.property.name,
+          location: { start, end },
+          expressionLocation: path.node.loc,
+          expression: getExpression(path)
+        });
+      }
+
+      if (t.isIdentifier(path)) {
+        const { start, end } = path.node.loc;
+
+        symbols.identifiers.push({
+          name: path.node.name,
+          expression: path.node.name,
+          location: { start, end }
+        });
+      }
+
+      if (t.isVariableDeclarator(path)) {
+        const node = path.node.id;
+        const { start, end } = path.node.loc;
+
+        symbols.identifiers.push({
+          name: node.name,
+          expression: node.name,
+          location: { start, end }
+        });
+      }
     }
   });
 
   symbolDeclarations.set(source.id, symbols);
   return symbols;
 }
+
+function addProperty(name, expression, path, prevPath) {
+  const computed = path && path.node.computed;
+  const prevComputed = prevPath && prevPath.node.computed;
+  const prevArray = t.isArrayExpression(prevPath);
+  const array = t.isArrayExpression(path);
+
+  if (expression === "") {
+    if (computed) {
+      return `[${name}]`;
+    }
+    return name;
+  }
+
+  if (computed || array) {
+    if (prevComputed || prevArray) {
+      return `[${name}]${expression}`;
+    }
+    return `[${name}].${expression}`;
+  }
+
+  if (prevComputed || prevArray) {
+    return `${name}${expression}`;
+  }
+
+  return `${name}.${expression}`;
+}
+
+function getMemberExpression(node, expression = "") {
+  if (t.isMemberExpression(node)) {
+    const name = node.property.name;
+
+    return getMemberExpression(node.object, addProperty(name, expression));
+  }
+
+  if (t.isCallExpression(node)) {
+    return "";
+  }
+
+  if (t.isThisExpression(node)) {
+    return `this.${expression}`;
+  }
+
+  if (t.isIdentifier(node)) {
+    return `${node.name}.${expression}`;
+  }
+
+  return expression;
+}
+
+function getObjectExpression(path, prevPath, expression = "") {
+  const name = path.node.key.name;
+
+  expression = addProperty(name, expression, path, prevPath);
+
+  prevPath = path;
+  path = path.parentPath && path.parentPath.parentPath;
+
+  return getExpression(path, prevPath, expression);
+}
+
+function getArrayExpression(path, prevPath, expression) {
+  const index = prevPath.parentPath.key;
+  expression = addProperty(index, expression, path, prevPath);
+
+  prevPath = path;
+  path = path.parentPath && path.parentPath.parentPath;
+
+  return getExpression(path, prevPath, expression);
+}
+
+function getExpression(path, prevPath, expression = "") {
+  if (t.isVariableDeclaration(path)) {
+    const node = path.node.declarations[0];
+    const name = node.id.name;
+    return addProperty(name, expression, path, prevPath);
+  }
+
+  if (t.isVariableDeclarator(path)) {
+    const node = path.node.id;
```

Comment:
again, a bit confused about why id is an node? would expect it to be a string...